src/kademlia/node_id.cpp

```diff
@@ -70,6 +70,19 @@ int distance_exp(node_id const& n1, node_id const& n2)
 	return (std::max)(159 - distance(n1, n2).count_leading_zeroes(), 0);
 }
 
+int min_distance_exp(node_id const& n1, std::vector<node_id> const& ids)
+{
+	if (ids.empty()) return 0;
+
+	std::vector<int> distances;
+	distances.reserve(ids.size());
+	std::transform(ids.begin()
+			, ids.end()
+			, std::back_inserter(distances)
+			, [&n1](node_id const& node_id) { return distance_exp(n1, node_id); });
+	return *std::min_element(distances.begin(), distances.end());
```

Comment:
In the future these operations can be chained, but there's no good support for that in the standard library today. boost has a transform_iterator which you could use. but it may be simpler to just write the loop manually too (which can still be pretty nice with the new range-for syntax)